src/TourCMS.php

```diff
@@ -22,385 +22,534 @@
-            		} else {
-		                $params .= "&single_tour_id=";
-            		}
-			
-			$params .= $tour;
-			
-		}
-
-		if($channel==0)
-			return($this->request('/p/hotels/search_range.xml'.$params));
-		else
-			return($this->request('/c/hotels/search_range.xml'.$params, $channel));
-	}
-
-	public function search_hotels_specific($params = "", $tour = "", $channel = 0) {
-		$params = $this->validateParams($params);
-
-		if (!empty($tour)) {
-            		
-			if (empty($params)) {
-                		$params = '?single_tour_id=';
-            		} else { 
-				$params .= "&single_tour_id=";
-			}
-			
-			$params .= $tour;
-			
-		}
-
-		if($channel==0)
-			return($this->request('/p/hotels/search_avail.xml'.$params));
-		else
-			return($this->request('/c/hotels/search_avail.xml'.$params, $channel));
-	}
-
-	public function list_product_filters($channel = 0) {
-			return($this->request('/c/tours/filters.xml', $channel));
-	}
-
-	public function update_tour($tour_data, $channel) {
-		return($this->request('/c/tour/update.xml', $channel, "POST", $tour_data));
-	}
-
-	public function update_tour_url($tour, $channel, $tour_url) {
-
-		$url_data = new SimpleXMLElement('<tour />');
-		$url_data->addChild('tour_id', $tour);
-		$url_data->addChild('tour_url', $tour_url);
-
-		return($this->update_tour($url_data, $channel));
-	}
-
-	public function list_tours($channel = 0, $params = "") {
-		$params = $this->validateParams($params);
-		if($channel==0)
-			return($this->request('/p/tours/list.xml'.$params));
-		else
-			return($this->request('/c/tours/list.xml'.$params, $channel));
-	}
-
-	public function list_tour_images($channel = 0, $params = "")
-	{
-		$params = $this->validateParams($params);
-		if($channel==0)
-			return($this->request('/p/tours/images/list.xml'.$params));
-		else
-			return($this->request('/c/tours/images/list.xml'.$params, $channel));
-	}
-
-	public function list_tour_locations($channel = 0, $params = "")
-	{
-		$params = $this->validateParams($params);
-		if($channel==0)
-			return($this->request('/p/tours/locations.xml'.$params));
-		else
-			return($this->request('/c/tours/locations.xml'.$params, $channel));
-	}
-
-	public function delete_tour($tour, $channel)
-	{
-		$url = '/c/tour/delete.xml?id='.$tour;
-		return($this->request($url, $channel, self::HTTP_VERB_POST));
-	}
-
-	public function show_tour($tour, $channel, $params = false)
-	{
-		$url = '/c/tour/show.xml?id='.$tour;
-
-		/*
+        // curl_setopt($ch, CURLOPT_CAINFO, "c:/path/to/ca-bundle.crt");
+
+        if ($verb == self::HTTP_VERB_POST) {
+            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, self::HTTP_VERB_POST);
+            if (!is_null($postData)) {
+                if (is_string($postData)) {
+                    curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);
+                }
+                if ($this->isXMLObject($postData)) {
+                    curl_setopt($ch, CURLOPT_POSTFIELDS, $postData->asXML());
+                }
```

Comment:
This code was old.
Could you remain using the instance of and remove that function please